python/social_bot/envs/grocery_ground.py

```diff
@@ -242,7 +391,7 @@ def reset(self):
         self._steps_in_this_episode = 0
         self._world.reset()
         self._teacher.reset(self._agent, self._world)
-        self._random_move_objects()
+        self._teacher_task.reset()
```

Comment:
should be called by teacher.

Reply:
Yes, should move some task operations to teacher, also including the building vocab procedure.